gwdetchar/io/html.py

```diff
@@ -68,6 +81,79 @@ def write_param(param, value):
     return page()
 
 
+def render_code(code, language):
+    """Render a black of code with syntax highlighting
+
+    Parameters
+    ----------
+    code : `str`
+        a raw block of source code
+
+    language : `str`
+        language the code is written in, e.g. `'python'`
+    """
+    lexer = get_lexer_by_name(language, stripall=True)
+    return highlight(code, lexer, FORMATTER)
+
+
+def get_command_line(language='bash'):
+    """Render the command line invocation used to generate a page
+
+    Parameters
+    ----------
+    language : `str`, optional
+        language the code is written in, default: `'bash'`
+    """
+    commandline = ' '.join(sys.argv)
+    return render_code(commandline, language)
+
+
+def write_footer(about=None, date=None, class_=False,
```

Comment:
I think this should be separated into two methods, one that generates the relevant string, and another that calls the first and wraps it into a `<footer>` element.

Those methods that don't want an actual `<footer>`, but just want the epilogue text can just call the first only.